collector/mysql_innodb_index_stats.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Scrape `mysql.innodb_index_stats`.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const mysqlIndexStatsQuery = `
+		  SELECT
+		    database_name,
+		    table_name,
+		    index_name,
+            stat_name,
+            stat_value
+		  FROM mysql.innodb_index_stats
+		`
+
+var (
+	indexStatLabelNames = []string{"database_name", "table_name", "index_name", "stat_name"}
+)
+
+// Metric descriptors.
+var (
+	indexStatsValueDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, mysql, "innodb_index_stats_stat_value"),
```

Comment:
This is going to make metrics named like `mysql_mysql_...`. I think we want `prometheus.BuildFQName(namespace, innodb, "index_stats_stat_value")`